app/models/vote.rb

```diff
@@ -0,0 +1,11 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+
+  validates :course_slug,
+    presence: true,
+    uniqueness: { scope: :user_id }
+
+  def self.get_course_vote_total(course_slug)
+    @total_votes = Vote.where(course_slug: course_slug).count
```

Comment:
Don't need the instance variable assignment here (unless you plan on memoizing it) – so either:

```ruby
Vote.where(course_slug: course_slug).count
```

… or:

```ruby
@total_votes ||= Vote.where(course_slug: course_slug).count
```

Reply:
Didn't memoize as we expect this to be updated frequently.